presto-main/src/main/java/com/facebook/presto/operator/scalar/TestingRowConstructor.java

```diff
@@ -152,17 +154,24 @@ public static Slice testNestedRow(@Nullable @SqlType(StandardTypes.DOUBLE) Doubl
 
     @ScalarFunction("test_row")
     @SqlType("row<timestamp>('col0')")
-    public static Slice testRowBigintBigint(@Nullable @SqlType(StandardTypes.TIMESTAMP) Long arg1)
+    public static Block testRowBigintBigint(@Nullable @SqlType(StandardTypes.TIMESTAMP) Long arg1)
     {
         return toStackRepresentation(ImmutableList.of(TIMESTAMP), arg1);
     }
 
-    private static Slice toStackRepresentation(List<Type> parameterTypes, Object... values)
+    public static Block toStackRepresentation(List<Type> parameterTypes, Object... values)
     {
-        BlockBuilder blockBuilder = new VariableWidthBlockBuilder(new BlockBuilderStatus(), 1024);
+        checkArgument(parameterTypes.size() == values.length, "parameterTypes.size(" + parameterTypes.size() + ") does not equal to values.length(" + values.length + ")");
+
+        ArrayBlockBuilder blockBuilder = new ArrayBlockBuilder(
```

Comment:
I think you can remove the `ArrayBlockBuilder` and just build the `InterleavedBlockBuilder` instead of getting the zeroth entry